### PR TITLE
Publish with the branch name

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: c2cciutils/scripts/publish.py [hunk 1]
```diff
@@ -20,10 +20,20 @@ def match(tpe, base_re):
     return re.match("^refs/{}/{}".format(tpe, base_re), os.environ["GITHUB_REF"])
 
 
+def to_version(full_config, value, kind):
+    item_re = c2cciutils.compile_re(full_config["version"].get(kind + "_to_version_re", []))
```
Comment (arnaud-morvan) [inline c2cciutils/scripts/publish.py:24]: Missing docstring
Ex : Compute publish version from branch name ot tag.

File: c2cciutils/scripts/publish.py [hunk 2]
```diff
@@ -20,10 +20,20 @@ def match(tpe, base_re):
...
+    if match[0] is not None:
+        return c2cciutils.get_value(*match)
+    return value
+
+
 def main() -> None:
     parser = argparse.ArgumentParser(description="Publish the project.")
     parser.add_argument("--group", default="default", help="The publishing group")
     parser.add_argument("--version", help="The version to publish to")
+    parser.add_argument("--branch", help="A publish for the branch")
```
Comment (arnaud-morvan) [inline c2cciutils/scripts/publish.py:35]: Not clear.

I would propose : The branch from which to compute the version.

But I do not really understand, is it not always possible to get the branch / tag from the current git repository ?
Before to publish we need to be on the considered branch / tag.

Comment (sbrunner) [reply c2cciutils/scripts/publish.py:35]: You say that I should use `git rev-parse --abbrev-ref HEAD`?
It's effectively a possibility...

Comment (arnaud-morvan) [reply c2cciutils/scripts/publish.py]: Yes, something like that, if you added this only for running publish local, it should be possible to the same locally as you do from GITHUB_REF.

Comment (arnaud-morvan) [reply c2cciutils/scripts/publish.py]: But locally it is possible to be on a branch and a tag, I suppose on GitHub you will get an independant workflow for each.

**Review by arnaud-morvan — APPROVED** [review]

**Pull request merged**